File: Олимпиады/Mana.cpp
#include <iostream>

using std::cout;
using std::cin;
using std::endl;


int surv(int mana, int* M, int n){
    for(int i = 0; i < n; i++){
        mana += M[i];
        if(mana < 0) return false;
    }
    return true;
}


int main(){
    int n = 0;
    cin >> n;
    int M[n];
    for(int i = 0; i < n; i++){
        cin >> M[i];
    }
    cout << endl;
    int m = 0;
    while(!surv(m, M, n)){
        m++;
    }
    cout << m;
    return 0;
}

File: Олимпиады/Mana.py

def surv(mana, M, n):
    for i in range(1, n):
        mana += M[i]
        if(mana < 0): return False
    return True


if __name__ == "__main__":
    n = int(input())
    M = []
    for i in range(1, n):
        M.append(int(input()))
    
    m = 0
    while(not surv(m, M, n)):
        m += 1
    print(m)




